source/sigv4.c

```diff
@@ -793,6 +854,236 @@ static SigV4Status_t getCredentialScope( SigV4Parameters_t * pSigV4Params,
     }
 
 /*-----------------------------------------------------------*/
+    static SigV4Status_t writeSignedHeaderToString( size_t headerIndex,
```

Comment:
As this function is only writing the "Signed Header" to the Canonical Request Buffer, it would be better to name is appropriately.
```suggestion
    static SigV4Status_t writeSignedHeaderToCanonicalRequest( size_t headerIndex,
```